File: LovePlay-UIKit/LovePlay/Class/Zone/View/DiscussListTopCell.m
//
//  DiscussListTopCell.m
//  LovePlay
//
//  Created by Yuns on 2017/2/3.
//  Copyright © 2017年 Yuns. All rights reserved.
//

#import "DiscussListTopCell.h"
#import "DiscussListModel.h"

@interface DiscussListTopCell ()
//UI
@property (nonatomic, strong) UILabel *topTextNode;
@property (nonatomic, strong) UILabel *titleTextNode;
@property (nonatomic, strong) UIView *underLineNode;
//Data
@property (nonatomic, strong) ForumThread *forumThread;
@end

@implementation DiscussListTopCell

+ (instancetype)cellWithTableView:(UITableView *)tableView
{
    static NSString *ID = @"DiscussListTopCell";
    DiscussListTopCell *cell = [tableView dequeueReusableCellWithIdentifier:ID];
    if (cell == nil) {
        cell = [[DiscussListTopCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:ID];
    }
    return cell;
}

- (instancetype)initWithStyle:(UITableViewCellStyle)style reuseIdentifier:(NSString *)reuseIdentifier
{
    self = [super initWithStyle:style reuseIdentifier:reuseIdentifier];
    if (self) {
        self.selectionStyle = UITableViewCellSelectionStyleNone;
        [self addSubNodes];
        [self mas_subViews];
    }
    return self;
}

#pragma mark - private
- (void)addSubNodes
{
    [self.contentView addSubview:self.topTextNode];

    [self.contentView addSubview:self.titleTextNode];

    [self.contentView addSubview:self.underLineNode];
}

- (void)mas_subViews
{
    [_topTextNode mas_makeConstraints:^(MASConstraintMaker *make) {
        make.width.mas_equalTo(45);
        make.height.mas_equalTo(25);
        make.top.equalTo(self.contentView).offset(10);
        make.left.equalTo(self.contentView).offset(10);
    }];
    
    [_titleTextNode mas_makeConstraints:^(MASConstraintMaker *make) {
        make.left.equalTo(_topTextNode.mas_right).offset(10);
        make.right.equalTo(self.contentView).offset(-10);
        make.centerY.equalTo(_topTextNode);
    }];
    
    [_underLineNode mas_makeConstraints:^(MASConstraintMaker *make) {
        make.height.mas_equalTo(0.5);
        make.top.equalTo(_topTextNode.mas_bottom).offset(10);
        make.left.bottom.right.equalTo(self.contentView);
    }];
}

#pragma mark - public
- (void)setupForumThread:(ForumThread *)forumThread
{
    _forumThread = forumThread;
    _titleTextNode.text = forumThread.subject;
}

#pragma mark - setter / getter
- (UILabel *)topTextNode
{
    if (!_topTextNode) {
        UILabel *topTextNode = [[UILabel alloc] init];
        topTextNode.font = [UIFont systemFontOfSize:12];
        topTextNode.textAlignment = NSTextAlignmentCenter;
        topTextNode.textColor = RGB(236, 126, 150);
        topTextNode.layer.cornerRadius = 5;
        topTextNode.layer.masksToBounds = YES;
        topTextNode.layer.borderWidth = 1;
        topTextNode.layer.borderColor = RGB(236, 126, 150).CGColor;
        topTextNode.text = @"置顶帖";
        _topTextNode = topTextNode;
    }
    return _topTextNode;
}

- (UILabel *)titleTextNode
{
    if (!_titleTextNode) {
        UILabel *titleTextNode = [[UILabel alloc] init];
        titleTextNode.font = [UIFont systemFontOfSize:14];
        titleTextNode.textColor = RGB(36, 36, 36);
        _titleTextNode = titleTextNode;
    }
    return _titleTextNode;
}

- (UIView *)underLineNode
{
    if (!_underLineNode) {
        UIView *underLineNode = [[UIView alloc] init];
        underLineNode.backgroundColor = RGB(222, 222, 222);
        _underLineNode = underLineNode;
    }
    return _underLineNode;
}

@end

File: LovePlay-Swift3/LovePlay/Class/Zone/View/DiscussListTopCell.swift
//
//  DiscussListTopCell.swift
//  LovePlay
//
//  Created by weiying on 2017/6/6.
//  Copyright © 2017年 yuns. All rights reserved.
//

import UIKit

class DiscussListTopCell: UITableViewCell {

    class func cellWithTableView(tableView : UITableView) -> DiscussListTopCell {
        let ID = NSStringFromClass(self)
        var cell : DiscussListTopCell? = tableView.dequeueReusableCell(withIdentifier: ID) as? DiscussListTopCell
        if cell == nil {
            cell = DiscussListTopCell(style: .default, reuseIdentifier: ID)
        }
        return cell!
    }
    
    override init(style: UITableViewCellStyle, reuseIdentifier: String?) {
        super.init(style: style, reuseIdentifier: reuseIdentifier)
        self.selectionStyle = .none
        self.addSubViews()
        self.snp_subViews()
    }
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    
    // MARK: - private
    private func addSubViews() {
        self.contentView.addSubview(self.topTextLabel)
        self.contentView.addSubview(self.titleTextLabel)
        self.contentView.addSubview(self.underLineView)
    }
    
    private func snp_subViews() {
        self.topTextLabel.snp.makeConstraints { (make) in
            make.height.equalTo(25)
            make.width.equalTo(45)
            make.top.equalToSuperview().offset(10)
            make.left.equalToSuperview().offset(10)
        }
        
        self.titleTextLabel.snp.makeConstraints { (make) in
            make.left.equalTo(self.topTextLabel.snp.right).offset(10)
            make.right.equalToSuperview().offset(-10)
            make.centerY.equalTo(self.topTextLabel)
        }
        
        self.underLineView.snp.makeConstraints { (make) in
            make.height.equalTo(0.5)
            make.top.equalTo(self.topTextLabel.snp.bottom).offset(10)
            make.left.bottom.right.equalToSuperview()
        }
    }
    
    fileprivate var _listModel : DiscussListModel?
    var listModel : DiscussListModel? {
        set {
            _listModel = newValue
        }
        
        get {
            return _listModel
        }
    }
    
    //MARK: - setter / getter
    lazy var topTextLabel : UILabel = {
        let label : UILabel = UILabel()
        label.font = UIFont.systemFont(ofSize: 12)
        label.textColor = RGB(r: 236, g: 126, b: 150)
        return label
    }()
    
    lazy var titleTextLabel : UILabel = {
        let label : UILabel = UILabel()
        label.font = UIFont.systemFont(ofSize: 14)
        label.textColor = RGB(r: 36, g: 36, b: 36)
        return label
    }()
    
    lazy var underLineView : UIView = {
        let lineView : UIView = UIView()
        lineView.backgroundColor = RGB(r: 222, g: 222, b: 222)
        return lineView
    }()
}
